### ISSUE TITLE
Orphaned named actor implementation plan

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#5957]
In ray serve, we need an actor that handles that can survive after driver exits. Ray serve put all actor actors inside a single actor called ActorNursery. The ActorNursery is a single actor inside the ray cluster accessible from multiple driver. Right now, this is done by use a num_cpu=0 tasks to hold the actor handle for ActorNursery in a `while True` loop. 

But this is hack. What we really need is to have **a named actor that does not get destroyed when go out of scope**.

Implementation plan:
Python:
- `ray.experimental.named_actor.register_actor` will set `actor_handle._ray_original_handle = False` so `actor_handle.__del__` won't be called. 

Raylet:
- Implement a new RayletClient call `MarkActorOrphaned`. Will closely follows https://github.com/ray-project/ray/blob/bc4a0de4daaac3b5f718f1c15fb6fd70f48f6f17/src/ray/raylet/raylet_client.cc#L356
- Add Flatbuffer message
 ```
table MarkActorOrphanedRequest {
  // ID of the actor.
  actor_id: string;
}
```
- The node manager implementation of `MarkActorOrphaned` will set an attribute in the worker object `is_orphaned_` as `true`. 
- Added fields to the `Worker class`:
```
// Indicate whether the actor should not be destroyed on driver exit
bool is_orphaned_;
```
- On driver exit, modify the code that terminates all workers under that driver https://github.com/ray-project/ray/blob/bc4a0de4daaac3b5f718f1c15fb6fd70f48f6f17/src/ray/raylet/node_manager.cc#L265-L272 such that if `worker.is_orphaned_` will not be terminated.

**Comment by ericl** [comment]
Can we just make this an attribute of creating the actor instead of adding a new call?

**Comment by stephanie-wang** [comment]
Agree with Eric that we should do it when creating the actor, ideally in [this struct](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/common.h#L101). Otherwise, looks good, although it's likely we'll have to move this code from the raylet to the core worker soon enough.

**Comment by simon-mo** [comment]
Ok sounds good. I will do it when creating the actor. I'll ping you once the PR is created.

**Comment by simon-mo** [comment]
Fixed by #6036
